cpu/stm32/periph/gpio_all.c

```diff
@@ -338,8 +339,8 @@ void isr_exti(void)
 #if defined(CPU_FAM_STM32G0) || defined(CPU_FAM_STM32L5) || \
     defined(CPU_FAM_STM32MP1)
     /* only generate interrupts against lines which have their IMR set */
-    uint32_t pending_rising_isr = (EXTI->RPR1 & EXTI_REG_IMR);
-    uint32_t pending_falling_isr = (EXTI->FPR1 & EXTI_REG_IMR);
+    uint32_t pending_rising_isr = (EXTI->RPR1 & EXTI_REG_IMR & EXTI_MASK);
+    uint32_t pending_falling_isr = (EXTI->FPR1 & EXTI_REG_IMR & EXTI_MASK);
```

Comment:
Just checked, we really don't even enable anything above `EXTI15_IRQn` and `_pin_num()` is always capped at 0xF